### ISSUE TITLE
Tabnine 2.8.7 binary not downloaded Inside Proxy and tool doesnt work.

### ISSUE THREAD

**Issue by sannamalai76** [svipas/vscode-code-autocomplete#2]
I tried to install the tool inside the proxy network and tabnine 2.8.7 is not downloaded and getting a message -saying extension is dead.

**Comment by svipas** [comment]
I didn't thought about this one and to be honest I forked and worked on this extension to meet some of my needs. You can manually download TabNine binary from official extension repo ([here](https://github.com/codota/tabnine-vscode/tree/master/binaries/2.8.7)).

**Comment by sannamalai76** [comment]
Thanks for the response, where should i copy the binary

**Comment by svipas** [comment]
Download all the binaries it would be easier instead of identifying which arch, platform, etc. you're using and place it inside extensions dir. I'm using macOS so it should be here: `/Users/svipas/.vscode/extensions/svipas.code-autocomplete-1.1.0/binaries`

**Comment by svipas** [comment]
@sannamalai76  You said you're using proxy network so as far as I understand it blocks network request to download TabNine binary? If so maybe I should update `README` or add a command which points you to binaries, etc.

**Comment by sannamalai76** [comment]
Thanks svipas, Yes it will be a good idea to update the read me , there are lot of folks like me want to use inside proxy

**Comment by sannamalai76** [comment]
I copied the file but it keeps trying to download everytime i open code. is there any settings i need to update after i copy the binary file

**Comment by svipas** [comment]
Yeah, I know it contains problems with downloading TabNine binary which I will fix. So basically what I'm thinking is:

- Use different path for TabNine binary. This would make easier to place it if auto download fails for e.g. you're using Proxy, etc. And most importantly it wouldn't download TabNine binary if extension is updated, because right now binary is placed inside extension dir.
- Update `README` with info how to download binaries manually and where to place them.
- If download fails show notification which points to the `README` of how to download it manually, etc.
- If binary already exists in path don't download TabNine again. I'm saving version in VS Code storage, I will need to change logic of the code which checks if TabNine exists.

@sannamalai76 I guess this covers all your problems?

**Comment by sannamalai76** [comment]
Yes it covers. thank you for doing this.

**Comment by sannamalai76** [comment]
Yes it covers. thank you for doing this., Please let me know when the next version will be available with the above fix
